File: day2/day2p2.py
file = open('input.txt', 'r+', encoding='utf-8')

lines = list()
line_array = list()

for line in file:
    lines.append(line.rstrip())

for line in lines:
    line_array.append(line.split('\t'))

file.close()

def process(line):
    temp = list()

    for num in line:
        temp.append(int(num))

    for num in temp:
        index = 0
        while index < len(temp):
            if num is temp[index]:
                index += 1
                continue
            if(num % temp[index]) == 0:
                return num / temp[index]
            index += 1

quotients = list()
for line in line_array:
    quotients.append(process(line))

checksum = 0
for num in quotients:
    checksum += num

print(f'Checksum: {checksum}')

File: day2/day2p2.js
const fs = require('fs');

fs.readFile('./input.txt', 'utf-8', (err, data) => {
    let numArray = split(data);
    console.log(`Checksum: ${calculate(numArray)}`);
});

const split = (data) => {
    let lines = data.split('\n');
    let nums = [];

    lines.forEach(line => {
        nums.push(line.trim().split('\t'));
    });

    for(let i = 0; i < nums.length; i++){
        for(let j = 0; j < nums[i].length; j++){
            nums[i][j] = parseInt(nums[i][j]);
        }
    }
    return nums;
};

// Ineligant but it works
const calculate = (nums) => {
    accumulator = 0;

    nums.forEach(line => {
        for(let i = 0; i < line.length; i++){
            for(let j = 0; j < line.length; j++){
                if(i === j){
                    continue;
                } else {
                    if((line[i] % line[j]) === 0){
                        accumulator += (line[i] / line[j]);
                    }
                }
            }
        }
    });

    return accumulator;
};